### PR TITLE
Code coverage

### PR DESCRIPTION
**Pull request by LouieVoit**
- This PR adds code coverage using gcov and lcov to generate an html summary file.
- A new Cmake target call coverage is added.
- Can only be enabled when CMAKE_BUILD_TYPE is Debug.
- The flag `_GLIBCXX_DEBUG` has been deleted in Debug mode since it was making the unit tests crashing with code coverage. We already had some issues with this flag in the past, I think it's safer to delete it for now.
- The workflow is as follow:
1. Compile the mamico tests: `make testmamico`
2. Run the tests: `ctest`
3. Then `make coverage` generates a coverage/index.html in the build directory

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,12 +10,18 @@ if(POLICY CMP0057)
...
+# Testing and coverage
 option(BUILD_TESTING "Build testing" OFF)
 message(STATUS ">> Build with testing: " ${BUILD_TESTING})
 if (BUILD_TESTING)
   include(CTest)
 endif()
+option(ENABLE_COVERAGE "Enable code coverage" OFF)
+message(STATUS ">> Enable code coverage: " ${ENABLE_COVERAGE})
+if (ENABLE_COVERAGE AND NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+  message(STATUS ">> Enable code coverage was enabled but CMAKE_BUILD_TYPE is not Debug, disabling code coverage ...")
```
Comment (rubenhorn) [inline CMakeLists.txt:22]: Could also be a warning IMHO

**Review by rubenhorn — APPROVED** [review]
:+1:   
I wonder if we could display the test coverage also in the output of the CI #11 (or maybe include as a build artifact)

**Comment by LouieVoit** [comment]
> 👍 I wonder if we could display the test coverage also in the output of the CI #11 (or maybe include as a build artifact)

Yes. The CI integration is planned :-)

File: CMakeLists.txt (hunk 1 above)
Comment (Thinkpiet) [reply CMakeLists.txt:22]: Done

**Pull request merged**